Fails to build with gcc 4.9.4

I am trying to build Valhalla 3.0.8 with relatively old gcc provideld by Sailfish OS. Looks like it cannot decipher one of the `auto` variables or if there is some other error. Help would be appreciated, even if just stating that its far below minimal required gcc version.

```
 [  2%] Building CXX object src/midgard/CMakeFiles/valhalla-midgard.dir/gridded_data.cc.o
cd /home/src1/osmscout/valhalla/pkg-valhalla-lite/valhalla/build-rpm/src/midgard && /usr/bin/c++  -DAUTO_DOWNLOAD=0 -DBOOST_NO_CXX11_SCOPED_ENUMS -DHAS_REMOTE_API=0 -DRAPIDJSON_HAS_STDSTRING -I/home/src1/osmscout/valhalla/pkg-valhalla-lite/valhalla -I/home/src1/osmscout/valhalla/pkg-valhalla-lite/valhalla/valhalla -I/home/src1/osmscout/valhalla/pkg-valhalla-lite/valhalla/build-rpm/src -I/home/src1/osmscout/valhalla/pkg-valhalla-lite/valhalla/build-rpm/src/valhalla  -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 -Wformat -Wformat-security -m32 -msse -msse2 -march=i686 -mfpmath=sse -mtune=generic -fno-omit-frame-pointer -fasynchronous-unwind-tables -fPIC -fdiagnostics-color=auto -O3 -DNDEBUG   -std=gnu++11 -o CMakeFiles/valhalla-midgard.dir/gridded_data.cc.o -c /home/src1/osmscout/valhalla/pkg-valhalla-lite/valhalla/src/midgard/gridded_data.cc
In file included from /home/src1/osmscout/valhalla/pkg-valhalla-lite/valhalla/src/midgard/gridded_data.cc:1:0:
/home/src1/osmscout/valhalla/pkg-valhalla-lite/valhalla/valhalla/midgard/gridded_data.h: In instantiation of ‘bool valhalla::midgard::GriddedData<coord_t>::Set(const coord_t&, float) [with coord_t = valhalla::midgard::Point2]’:
/home/src1/osmscout/valhalla/pkg-valhalla-lite/valhalla/src/midgard/gridded_data.cc:326:16:   required from here
/home/src1/osmscout/valhalla/pkg-valhalla-lite/valhalla/valhalla/midgard/gridded_data.h:40:33: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
     if (cell_id >= 0 && cell_id < data_.size()) {
                                 ^
/home/src1/osmscout/valhalla/pkg-valhalla-lite/valhalla/valhalla/midgard/gridded_data.h: In instantiation of ‘void valhalla::midgard::GriddedData<coord_t>::SetIfLessThan(int, float) [with coord_t = valhalla::midgard::Point2]’:
/home/src1/osmscout/valhalla/pkg-valhalla-lite/valhalla/src/midgard/gridded_data.cc:326:16:   required from here
/home/src1/osmscout/valhalla/pkg-valhalla-lite/valhalla/valhalla/midgard/gridded_data.h:54:33: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
     if (tile_id >= 0 && tile_id < data_.size() && value < data_[tile_id]) {
                                 ^
/home/src1/osmscout/valhalla/pkg-valhalla-lite/valhalla/valhalla/midgard/gridded_data.h: In instantiation of ‘void valhalla::midgard::GriddedData<coord_t>::SetIfLessThan(const coord_t&, float) [with coord_t = valhalla::midgard::Point2]’:
/home/src1/osmscout/valhalla/pkg-valhalla-lite/valhalla/src/midgard/gridded_data.cc:326:16:   required from here
/home/src1/osmscout/valhalla/pkg-valhalla-lite/valhalla/valhalla/midgard/gridded_data.h:68:33: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
     if (cell_id >= 0 && cell_id < data_.size() && value < data_[cell_id]) {
                                 ^
/home/src1/osmscout/valhalla/pkg-valhalla-lite/valhalla/src/midgard/gridded_data.cc: In instantiation of ‘valhalla::midgard::GriddedData<coord_t>::contours_t valhalla::midgard::GriddedData<coord_t>::GenerateContours(const std::vector<float>&, bool, float, float) const [with coord_t = valhalla::midgard::Point2; valhalla::midgard::GriddedData<coord_t>::contours_t = std::map<float, std::list<std::list<std::list<valhalla::midgard::Point2, std::allocator<valhalla::midgard::Point2> >, std::allocator<std::list<valhalla::midgard::Point2, std::allocator<valhalla::midgard::Point2> > > >, std::allocator<std::list<std::list<valhalla::midgard::Point2, std::allocator<valhalla::midgard::Point2> >, std::allocator<std::list<valhalla::midgard::Point2, std::allocator<valhalla::midgard::Point2> > > > > >, std::function<bool(float, float)>, std::allocator<std::pair<const float, std::list<std::list<std::list<valhalla::midgard::Point2, std::allocator<valhalla::midgard::Point2> >, std::allocator<std::list<valhalla::midgard::Point2, std::allocator<valhalla::midgard::Point2> > > >, std::allocator<std::list<std::list<valhalla::midgard::Point2, std::allocator<valhalla::midgard::Point2> >, std::allocator<std::list<valhalla::midgard::Point2, std::allocator<valhalla::midgard::Point2> > > > > > > > >]’:
/home/src1/osmscout/valhalla/pkg-valhalla-lite/valhalla/src/midgard/gridded_data.cc:326:16:   required from here
/home/src1/osmscout/valhalla/pkg-valhalla-lite/valhalla/src/midgard/gridded_data.cc:301:60: error: converting to ‘const std::unordered_set<unsigned int>’ from initializer list would use explicit constructor ‘std::unordered_set<_Value, _Hash, _Pred, _Alloc>::unordered_set(std::unordered_set<_Value, _Hash, _Pred, _Alloc>::size_type, const hasher&, const key_equal&, const allocator_type&) [with _Value = unsigned int; _Hash = std::hash<unsigned int>; _Pred = std::equal_to<unsigned int>; _Alloc = std::allocator<unsigned int>; std::unordered_set<_Value, _Hash, _Pred, _Alloc>::size_type = unsigned int; std::unordered_set<_Value, _Hash, _Pred, _Alloc>::hasher = std::hash<unsigned int>; std::unordered_set<_Value, _Hash, _Pred, _Alloc>::key_equal = std::equal_to<unsigned int>; std::unordered_set<_Value, _Hash, _Pred, _Alloc>::allocator_type = std::allocator<unsigned int>]’
         Polyline2<coord_t>::Generalize(line, gen_factor, {});
                                                            ^
In file included from /home/src1/osmscout/valhalla/pkg-valhalla-lite/valhalla/src/midgard/gridded_data.cc:1:0:
/home/src1/osmscout/valhalla/pkg-valhalla-lite/valhalla/valhalla/midgard/gridded_data.h: In instantiation of ‘bool valhalla::midgard::GriddedData<coord_t>::Set(const coord_t&, float) [with coord_t = valhalla::midgard::PointLL]’:
/home/src1/osmscout/valhalla/pkg-valhalla-lite/valhalla/src/midgard/gridded_data.cc:327:16:   required from here
/home/src1/osmscout/valhalla/pkg-valhalla-lite/valhalla/valhalla/midgard/gridded_data.h:40:33: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
     if (cell_id >= 0 && cell_id < data_.size()) {
                                 ^
/home/src1/osmscout/valhalla/pkg-valhalla-lite/valhalla/valhalla/midgard/gridded_data.h: In instantiation of ‘void valhalla::midgard::GriddedData<coord_t>::SetIfLessThan(int, float) [with coord_t = valhalla::midgard::PointLL]’:
/home/src1/osmscout/valhalla/pkg-valhalla-lite/valhalla/src/midgard/gridded_data.cc:327:16:   required from here
/home/src1/osmscout/valhalla/pkg-valhalla-lite/valhalla/valhalla/midgard/gridded_data.h:54:33: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
     if (tile_id >= 0 && tile_id < data_.size() && value < data_[tile_id]) {
                                 ^
/home/src1/osmscout/valhalla/pkg-valhalla-lite/valhalla/valhalla/midgard/gridded_data.h: In instantiation of ‘void valhalla::midgard::GriddedData<coord_t>::SetIfLessThan(const coord_t&, float) [with coord_t = valhalla::midgard::PointLL]’:
/home/src1/osmscout/valhalla/pkg-valhalla-lite/valhalla/src/midgard/gridded_data.cc:327:16:   required from here
/home/src1/osmscout/valhalla/pkg-valhalla-lite/valhalla/valhalla/midgard/gridded_data.h:68:33: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
     if (cell_id >= 0 && cell_id < data_.size() && value < data_[cell_id]) {
                                 ^
/home/src1/osmscout/valhalla/pkg-valhalla-lite/valhalla/src/midgard/gridded_data.cc: In instantiation of ‘valhalla::midgard::GriddedData<coord_t>::contours_t valhalla::midgard::GriddedData<coord_t>::GenerateContours(const std::vector<float>&, bool, float, float) const [with coord_t = valhalla::midgard::PointLL; valhalla::midgard::GriddedData<coord_t>::contours_t = std::map<float, std::list<std::list<std::list<valhalla::midgard::PointLL> >, std::allocator<std::list<std::list<valhalla::midgard::PointLL> > > >, std::function<bool(float, float)>, std::allocator<std::pair<const float, std::list<std::list<std::list<valhalla::midgard::PointLL> >, std::allocator<std::list<std::list<valhalla::midgard::PointLL> > > > > > >]’:
/home/src1/osmscout/valhalla/pkg-valhalla-lite/valhalla/src/midgard/gridded_data.cc:327:16:   required from here
/home/src1/osmscout/valhalla/pkg-valhalla-lite/valhalla/src/midgard/gridded_data.cc:301:60: error: converting to ‘const std::unordered_set<unsigned int>’ from initializer list would use explicit constructor ‘std::unordered_set<_Value, _Hash, _Pred, _Alloc>::unordered_set(std::unordered_set<_Value, _Hash, _Pred, _Alloc>::size_type, const hasher&, const key_equal&, const allocator_type&) [with _Value = unsigned int; _Hash = std::hash<unsigned int>; _Pred = std::equal_to<unsigned int>; _Alloc = std::allocator<unsigned int>; std::unordered_set<_Value, _Hash, _Pred, _Alloc>::size_type = unsigned int; std::unordered_set<_Value, _Hash, _Pred, _Alloc>::hasher = std::hash<unsigned int>; std::unordered_set<_Value, _Hash, _Pred, _Alloc>::key_equal = std::equal_to<unsigned int>; std::unordered_set<_Value, _Hash, _Pred, _Alloc>::allocator_type = std::allocator<unsigned int>]’
         Polyline2<coord_t>::Generalize(line, gen_factor, {});
                                                            ^
make[2]: *** [src/midgard/CMakeFiles/valhalla-midgard.dir/build.make:92: src/midgard/CMakeFiles/valhalla-midgard.dir/gridded_data.cc.o] Error 1
make[2]: Leaving directory '/home/src1/osmscout/valhalla/pkg-valhalla-lite/valhalla/build-rpm'
make[1]: *** [CMakeFiles/Makefile2:847: src/midgard/CMakeFiles/valhalla-midgard.dir/all] Error 2
make[1]: Leaving directory '/home/src1/osmscout/valhalla/pkg-valhalla-lite/valhalla/build-rpm'
make: *** [Makefile:155: all] Error 2
```

Looks like gcc-5 is required at least:

```
/home/src1/osmscout/valhalla/pkg-valhalla-lite/valhalla/valhalla/baldr/datetime.h:188:9: error: ‘get_time’ is not a member of ‘std’
   ss >> std::get_time(&t, "%Y-%m-%dT%H:%M");
```

According to https://www.gnu.org/software/gcc/gcc-5/changes.html , it introduced `get_time`

--- a/README.md
+++ b/README.md
@@ -84,7 +84,7 @@ sudo apt-get install -y valhalla-bin
 Building from Source
 --------------------
 
-Valhalla uses CMake as build system.
+Valhalla uses CMake as build system. When compiling with gcc (GNU Compiler Collection), version 5 or newer is supported.
 
 To install on a Debian or Ubuntu system you need to install its dependencies with:
 

